### PR TITLE
Issue #6703: split JavadocMethod for missing javadocs

### PR DESCRIPTION
**Pull request by rnveach**
Issue #6703

Some properties are not really used anymore, but remain to not break compatibility.
This also affected the google config.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_JAVADOC_MISSING = "javadoc.missing";
+
+    /** Default value of minimal amount of lines in method to allow no documentation.*/
+    private static final int DEFAULT_MIN_LINE_COUNT = -1;
+
+    /** Specify the visibility scope where Javadoc comments are checked. */
+    private Scope scope = Scope.PRIVATE;
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java]: @romani Do you want the scope to be PUBLIC just like the other check?

**Comment by rnveach** [comment]
@romani Do we want to handle this PR/check special since it is also breaking compatibility or do you want me to restore the properties and create a new issue to remove them?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java]: yes, even scope type is very bad stuff to my mind.
Checks should be reasonable by default behavior, nobody writes javadoc in modern projects on non-API (public) methods.

**Comment by romani** [comment]
> Some properties are not really used anymore, but remain to not break compatibility.

you can put a comment on them and mark them as deprecated or ...

> This also affected the google config.

just launch diff regression testing .... if no diff ... we safe to merge.

-----
do not forget to update issue as breaking compatibilty, and provide migration notes to users. It will be a mess to upgrade in plugins and at end users.

**Comment by rnveach** [comment]
This PR should be updated to be issue #6703

**Comment by rnveach** [comment]
> you can put a comment on them and mark them as deprecate

Done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java (hunk 1 above)
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java]: Done.

**Comment by rnveach** [comment]
> do not forget to update issue as breaking compatibilty, and provide migration notes to users. 

Done.

> just launch diff regression testing .... if no diff ... we safe to merge.

Regression: http://rveach.no-ip.org/checkstyle/regression/reports/246/

There are a lot of differences because the check name changed, but the messages are all the same.
I checked and few and saw no issues.

**Pull request merged**